### DISCUSSION TITLE
spotify liked songs as playlist

### DISCUSSION THREAD

**Discussion by stijnverwaest** [music-assistant/support#496] [Feature requests and ideas]
Hello, first off i love your integration thanks for making it 

I would love if it was possible to add the Spotify liked songs as a playlist.
To have it start with the most recently liked song and continue playing the list.
Not sure if it would be possible or if their is enough demand for it .

Thanks for your time :)

**Comment by Paytrix** [comment]
I would love to see that as well! 
Been missing that in a lot of Spotify Integrations/Cards in HomeAssistant.
There have been a few Lovelace Cards which supported the liked songs library as a selectable playlist, so that should be something that's possible! :) 

For the meantime, I use an ifttt applet that automatically adds a new liked song into a playlist of your choice. 
With that you can select that playlist and play it back. The problem is, you can't sort the titles by "recently added", which would be handy as well!

(My iftt is kinda not working anymore... they change a lot and now have a premium service etc... I just gave it another try, maybe it works again)

**Reply by timvdsm** [reply]
That would be really great!
What existing lovelace card support it if I may ask?

**Comment by Albuca** [comment]
Bumping this, as there is a API endpoint that exposes a Spotify user's Liked Songs/Saved Tracks:  https://developer.spotify.com/console/get-current-user-saved-tracks

Generating an OAuth Token and trying it on the Spotify dev panel yields valid results. Unfortunately, I do not know how to translate the output json into something music-assistant can use to generate a playlist.

**Reply by OzGav** [reply]
Dev time is short but this sort of thing is on the todo list.

**Comment by stijnverwaest** [comment]
thanks !
